opensearch/sycamore-opensearch.sh

```diff
@@ -423,15 +390,138 @@ deploy_model() {
     if [[ -z "${model_id}" ]]; then
         wait_task "${task_id}" "$name model to become ready"
         model_id="${GET_TASK_MODEL_ID}"
-        echo "Fetched $name model id $model_id for task_id $task_id"
+        info "Fetched $name model id $model_id for task_id $task_id"
     else
-        echo "Using existing model id ${model_id}"
+        info "Using existing model id ${model_id}"
     fi
-    MODEL_ID="${model_id}"
-    DEPLOY_MODEL_LOG_FILE="${ARYN_STATUSDIR}/curl.deploy_model_task.${name}"
-    wait_or_die deploy_model_try "deploy of model ${MODEL_ID}, task ${task_id}, name ${name}" 60
-    wait_task "${DEPLOY_MODEL_DEP_TASK_ID}" "$name model to deploy"
-    # TODO: https://github.com/aryn-ai/sycamore/issues/153 - debug task waiting
+
+    if model_is_deployed "${model_id}"; then
+        info "Model $1 for $3 already deployed"
+        return 0
+    fi
+
+    # Create deploy task
+    local deploy_model_log_file="${ARYN_STATUSDIR}/curl.better_deploy_model_task.${name}"
+    spawn_deploy_model_task "${name}" "${model_id}"
+    local deploy_task_id="$(jq -r '.task_id' "${deploy_model_log_file}")"
+
+    # Cycle on the task status
+    debug "Wait for deploy task to finish"
+    local deploy_status_file="${ARYN_STATUSDIR}/curl.deploy_task_status.${name}"
+    local deploy_task_search_file="${ARYN_STATUSDIR}/curl.deploy_task_search_result.${name}"
+    local i
+    local max_reps=60
+    for i in $(seq "${max_reps}"); do
+        get_deploy_status_and_act_on_it "${deploy_task_id}" "${name}" "${model_id}" "${i}" "${max_reps}" && return 0
+        deploy_task_id="$(jq -r '.task_id' "${deploy_model_log_file}")"
+    done
+    die "Out of time to deploy model ${name}"
+}
+
+get_deploy_status_and_act_on_it() {
+    local deploy_task_id="$1"
+    local name="$2"
+    local model_id="$3"
+    local i="$4"
+    local max_reps="$5"
+    local deploy_model_log_file="${ARYN_STATUSDIR}/curl.better_deploy_model_task.${name}"
+
+    _curl "${BASE_URL}/_plugins/_ml/tasks/${deploy_task_id}" -o "${deploy_status_file}"
+    local status="$(jq -r '.state' "${deploy_status_file}")"
+    debug "${status}"
+    # Case 1: RUNNING / state not found. Task is still running so wait.
+    if [[ "${status}" == 'null' || "${status}" == 'RUNNING' || "${status}" == 'CREATED' ]]; then
+        info "Waiting for ${name} to deploy... ${i}/${max_reps}"
+        sleep 1
```

Comment:
Can we move the sleep to L416 above?  That would put the higher-level control all in one place.